src/SAWScript/Prover/Exporter.hs

```diff
@@ -125,6 +134,42 @@ writeAIG proxy sc f t = do
     aig <- bitblastPrim proxy sc t
     AIG.writeAiger f aig
 
+withABCVerilog :: SharedContext -> FilePath -> Term -> (FilePath -> String) -> TopLevel ()
+withABCVerilog sc baseName t buildCmd =
+  do verilogFile <- liftIO $ emptySystemTempFile (baseName ++ ".v")
+     write_verilog sc verilogFile t
+     liftIO $
+       do (out, err) <- readProcessExitIfFailure "abc" ["-q", buildCmd verilogFile]
+          unless (null out) $ putStrLn "ABC output:" >> putStrLn out
+          unless (null err) $ putStrLn "ABC errors:" >> putStrLn err
+          removeFile verilogFile
+
+-- | Write a @SATQuery@ to an AIG file by using ABC to convert a Verilog
+-- file.
+writeAIG_SATviaVerilog :: SharedContext -> FilePath -> SATQuery -> TopLevel ()
+writeAIG_SATviaVerilog sc f query =
+  writeAIGviaVerilog sc f =<< liftIO (satQueryAsTerm sc query)
+
+-- | Write a @Term@ representing a an arbitrary function to an AIG file
+-- by using ABC to convert a Verilog file.
+writeAIGviaVerilog :: SharedContext -> FilePath -> Term -> TopLevel ()
+writeAIGviaVerilog sc aigFile t =
+  withABCVerilog sc aigFile t $
+      \verilogFile -> "%read " ++ verilogFile ++ "; %blast; &write " ++ aigFile
```

Comment:
This looks like it might be fragile with respect to files with special characters in their name, etc.  Should we be doing some more careful escaping here?

Reply:
That's a good point. I'm not even sure to what extent ABC supports escaping. I don't suspect it's urgent, but I'm going to create an issue for this and other cases where we build compound file names or command lines and might want to be a little more careful.

Reply:
Created #1199 for this.